src/xunit.performance.api/Profilers/Etw/EtwPerfInfoPMCSample.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Xunit.Performance.Api.Profilers.Etw
+{
+    internal sealed class EtwPerfInfoPMCSample
+    {
+        public int ProcessId { get; set; }
+
+        public ulong InstructionPointer { get; set; }
+
+        public int ProfileSource { get; set; }
+
+        public int ThreadId { get; set; }
+
+        public DateTime TimeStamp { get; set; }
```

Comment:
You should check - I think that using a DateTime here may be costly - you might want to convert this on-the-fly from a QPC timestamp if it is ever requested.

Reply:
Looks like you may not have a choice here after all...

---
In reply to: [143034936](https://github.com/Microsoft/xunit-performance/pull/241#discussion_r143034936) [](ancestors = 143034936)